app.go

```diff
@@ -172,3 +179,12 @@ func funcLocation(c interface{}) (string, int) {
 	mfunc := runtime.FuncForPC(reflect.ValueOf(c).Pointer())
 	return mfunc.FileLine(mfunc.Entry())
 }
+
+func logCaller(name string) {
+	_, file, line, ok := runtime.Caller(2)
```

Comment:
If we want to go down this direction, need to remove the top-most call as it will always be app.go